src/sentry/integrations/github/integration.py

```diff
@@ -311,6 +311,7 @@ def search_issues(self, query: str | None, **kwargs) -> dict[str, Any]:
 class GitHubIntegrationProvider(IntegrationProvider):
     key = "github"
     name = "GitHub"
+    domain = IntegrationDomain.SOURCE_CODE_MANAGEMENT
```

Comment:
This is definitely a bit nicer. Tying the domain to the provider class really solidifies what an integration's responsibility is at instantiation time. I also remember @ameliahsu mentioning turning this into an array, which should work for the multiple domains on a single integration problem, but is the issue that we need a single "primary" domain per integration?